pkg/daemons/agent/agent.go

```diff
@@ -72,7 +71,7 @@ func startKubelet(ctx context.Context, cfg *daemonconfig.Agent) error {
 		return pkgerrors.WithMessage(err, "prepare user configuration drop-ins")
 	}
 
-	if err := writeKubeletConfig(cfg.KubeletConfigDir, defaultConfig); err != nil {
+	if err := writeKubeletConfig(cfg.KubeletConfig, defaultConfig); err != nil {
 		return pkgerrors.WithMessage(err, "generate default kubelet configuration drop-in")
```

Comment:
Same change for the message on line 66

```suggestion
		return pkgerrors.WithMessage(err, "generate default kubelet configuration")
```